From 791d72a58c12d44c6cc5a310b63bc5190e9964d0 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Thu, 2 Jun 2022 17:47:10 +0200
Subject: [PATCH] make: RISC-V does not support -march=native

-march=native is not an allowable argument for GCC on RISC-V.

Not providing -march will let GCC fallback to its built in default.
This is preferable to specifying a safe fixed value like -march=rv64gc.

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Change-Id: I5e11179323b005f330be4752012285451e791205
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/spdk/+/12880
Community-CI: Broadcom CI <spdk-ci.pdl@broadcom.com>
Community-CI: Mellanox Build Bot
Reviewed-by: Jim Harris <james.r.harris@intel.com>
Reviewed-by: Aleksey Marchuk <alexeymar@nvidia.com>
Tested-by: SPDK CI Jenkins <sys_sgci@intel.com>
---
 mk/spdk.common.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mk/spdk.common.mk b/mk/spdk.common.mk
index 37e0ca43d..c57a89129 100644
--- a/mk/spdk.common.mk
+++ b/mk/spdk.common.mk
@@ -88,6 +88,8 @@ COMMON_CFLAGS += -mcpu=$(TARGET_ARCHITECTURE)
 else ifeq ($(TARGET_MACHINE),aarch64)
 COMMON_CFLAGS += -march=$(TARGET_ARCHITECTURE)
 COMMON_CFLAGS += -DPAGE_SIZE=$(shell getconf PAGESIZE)
+else ifeq ('$(TARGET_MACHINE)|$(TARGET_ARCHITECTURE)','riscv64|native')
+# -march=native is not yet supported by GCC on RISC-V. Falling back to default.
 else
 COMMON_CFLAGS += -march=$(TARGET_ARCHITECTURE)
 endif
-- 
2.36.1

